### ISSUE TITLE
Create a flag emptyGetsAll

### ISSUE THREAD

**Issue by aworld1** [krisk/Fuse#728]
### Is there an existing issue or pull request for this?

- [X] I have searched the existing issues and pull requests

### Feature description

Create an option that when the query is empty, instead of getting no results, all results are returned.

### Desired solution

non-empty query: current behavior
emptyGetsAll is set to true: all results returned on empty input
emptyGetsAll is set to false: no results returned on empty input
default for flag should be false.

### Alternatives considered

Hardcoding outside of library; ugly.

### Additional context

The reason for this feature is if a search bar's behavior wants to get all entries on load when the query is empty (or when all characters are deleted), no hardcoded if-statement is necessary. Further, the items returned from Fuse.search are formatted differently than the list of objects fed as a parameter, as each object is placed as an "item" in the returned list. This improves consistency for the library and removes overhead.

**Comment by aworld1** [comment]
I have a branch ready with this feature, documentation, and tests, but am unable to push my branch because of permissions.

**Comment by jimmi-rbn** [comment]
Love the product but this feature is must have.

**Comment by stephane-klein** [comment]
Workaround in my Svelte project:

```
    $: if (fuse) {
        if (searchInput == "") {
            graphNodesStore.set(fuse._docs); // <= to get all results
        } else {
            graphNodesStore.set(fuse.search(searchInput).map((row) => row.item));
        }
    }
```

**Comment by krisk** [comment]
Fixed in d33b735f. An empty (or whitespace-only) string query now returns all docs as `FuseResult` objects with `item` and `refIndex`. The `limit` option is respected. No new flag needed.
